distribution/push.go

```diff
@@ -127,6 +133,19 @@ func Push(ctx context.Context, ref reference.Named, imagePushConfig *ImagePushCo
 			continue
 		}
 
+		parsedURL, urlErr := url.Parse(endpoint.URL)
```

Comment:
@calavera: I guess there would also be a method to mark a registry as a confirmed TLS registry? Where should the URL parsing happen for this? Should both methods have their own redundant calls to `url.Parse`?